pkg/rebuild/rebuild/rebuildremote.go

```diff
@@ -497,23 +498,21 @@ func RebuildRemote(ctx context.Context, input Input, id string, opts RemoteOptio
 	if err != nil {
 		return errors.Wrap(err, "creating build")
 	}
-	if err := doCloudBuild(ctx, opts.GCBClient, build, opts, &bi); err != nil {
-		return errors.Wrap(err, "performing build")
-	}
+	buildErr := errors.Wrap(doCloudBuild(ctx, opts.GCBClient, build, opts, &bi), "performing build")
 	{
 		lw, err := opts.LocalMetadataStore.Writer(ctx, Asset{Target: t, Type: BuildInfoAsset})
 		if err != nil {
-			return errors.Wrap(err, "creating writer for build info")
+			return stderrs.Join(buildErr, errors.Wrap(err, "creating writer for build info"))
 		}
 		defer lw.Close()
 		rw, err := opts.RemoteMetadataStore.Writer(ctx, Asset{Target: t, Type: BuildInfoAsset})
 		if err != nil {
-			return errors.Wrap(err, "creating remote writer for build info")
+			return stderrs.Join(buildErr, errors.Wrap(err, "creating remote writer for build info"))
 		}
 		defer rw.Close()
 		if err := json.NewEncoder(io.MultiWriter(lw, rw)).Encode(bi); err != nil {
-			return errors.Wrap(err, "marshalling and writing build info")
+			return stderrs.Join(buildErr, errors.Wrap(err, "marshalling and writing build info"))
```

Comment:
as discussed offline, errors.Join doesn't seem to add enough value to depart from the existing linear-error pattern we're using in the rebuild package. it might be good to break this out into an issue where we can look for opportunities to add apis that benefit from this structure. one issue brought up was that the top-level error no longer has the stack trace attached since it originates from std errors instead of pkg/errors.